tools/kagami/src/main.rs

```diff
@@ -849,13 +853,69 @@ mod validator {
         }
     }
 
+    pub fn clone_default_validator(out_dir: impl AsRef<Path>) -> Outcome {
+        const GIT_ORIGIN: &str = "https://github.com/hyperledger/iroha.git";
+
+        // Sparse directory copy as suggested in https://stackoverflow.com/a/52269934/10669935
+        cmd!(
+            "git",
+            "clone",
+            "-n",
+            "--depth=1",
+            "--filter=tree:0",
+            "--branch=iroha2-dev",
+            GIT_ORIGIN,
+            out_dir.as_ref()
+        )
```

Comment:
It is an issue for `kagami swarm --build-from-github`, because it clones the repo already, and it would be redundant to make it twice for validator building.